helix-core/src/syntax.rs

```diff
@@ -293,9 +302,25 @@ impl Loader {
                     .and_then(|extension| self.language_config_ids_by_file_type.get(extension))
             });
 
-        configuration_id.and_then(|&id| self.language_configs.get(id).cloned())
+        // If we have not found the configuration_id, see if we can get it from a shebang line
+        if configuration_id.is_none() {
+            if let Ok(mut file) = File::open(path) {
+                let mut buf = [0; 100];
+                if std::io::Read::read(&mut file, &mut buf[..]).is_ok() {
+                    if let Ok(str) = str::from_utf8(&buf) {
+                        static SHEBANG_REGEX: Lazy<Regex> = Lazy::new(|| {
+                            Regex::new(r"^#!\s*(?:\S*[/\\](?:env\s+)?)?([^\s\.\d]+)").unwrap()
+                        });
+                        configuration_id = SHEBANG_REGEX
+                            .captures(str)
+                            .and_then(|cap| cap.get(1))
+                            .and_then(|cap| self.language_config_ids_by_shebang.get(cap.as_str()))
+                    }
+                }
+            }
+        };
 
-        // TODO: content_regex handling conflict resolution
+        configuration_id.and_then(|&id| self.language_configs.get(id).cloned())
```

Comment:
I'd rather see this as a separate method, then called via `language_config_for_file_name(..).or_else(|| shebang(..))`